solutions/gold/cf-468B.mdx

```diff
@@ -0,0 +1,117 @@
+---
+id: cf-468B
+source: CF
+title: Two Sets
+author: Chongtian Ma
+---
+
+<Spoiler title="Hint 1">
+
+We can model the two sets as a graph, with $p_i$ and $a - p_i$ connected in graph $A$, and $p_i$ connected to $b - p_i$ in graph B.
+
+</Spoiler>
+
+<Spoiler title="Hint 2">
+
+How can there be a contradiction in the connected components?
+
+</Spoiler>
+
+## Explanation
+
+We can treat each set as a graph with several connected components. For each $p_i$, if $a - p_i$ exists, then $p_i$ and $a - p_i$ are in the same connected component inside graph $A$, and similarly for $B$. Each number in the connected component must belong to one set. Therefore, our condradiction comes from if there is a number $x$ in the component such that only $a - x$ exists, and another $y$ in the component such that only $b - y$ exists, or vice cersa.
+
+ We can check each component for the contradiction using DSU. The status of the whole component will be the intersection of what graph each number in the component can reside in.
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(N\log N)$
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+// BeginCodeSnip{DSU}
+struct DSU {
```

Comment:
use the module's dsu and i'll merge ❤️

Reply:
is this not benqs dsu?? certainly isnt mine

Reply:
what does this have to do with not wanting to use the modulel's dsu

Reply:
whats the module dsu?

Reply:
https://usaco.guide/gold/dsu#implementation

Reply:
wtf ok

Reply:
im never making another dsu internal sol agian